theano/tensor/nnet/nnet.py

```diff
@@ -2215,8 +2215,7 @@ def relu(x, alpha=0):
         return f1 * x + f2 * abs(x)
 
 
-def h_softmax(x, batch_size, n_outputs, n_classes, n_outputs_per_class,
-              W1, b1, W2, b2, target=None):
+def h_softmax(x, n_outputs, W1, b1, W2, b2, target=None):
```

Comment:
This break the interface... We don't like that. @adbrebs and @bouthilx What do you think of that? I saw one commit that changed the size to be in front instead of later parameter, what was the reason? Why you added those n_* that can be computed?

I see a few options, as this was merged the 11 Nov, so 3 mounts ago, we could:
1) Add extra parameter to have the same number of parameter with the same name, but different order:
h_softmax(x, n_outputs, W1, b1, W2, b2, batch_size=None, n_outputs=None, n_classes=None, n_outputs_per_class=None, target=None):

If the removed parameter are all used, then we raise an error. The user can have passed parameter by name or order and we don't know what he did.
2) Create a new function with a different name.
3) Use this interface:
def h_softmax(x, batch_size=None, n_outputs=None, n_classes=NOne, n_outputs_per_class=None,
             W1=None, b1=None, W2=None, b2=None, target=None):

Assert that W* b* aren't none. If some of the parameter you removed are none, use what you did, otherwise, use what the user passed.

Currently, I seem to prefer 3), but I would like to know what other people think of this. Maybe there is some cases where the shapes can't be used. If the user pass a constant, it could help get more constant in the graph.